pkg/client/client_test.go

```diff
@@ -1209,6 +1209,42 @@ var _ = Describe("Client", func() {
 				close(done)
 			})
 
+			It("should patch an existing object from a go struct, using optimistic locking", func(done Done) {
```

Comment:
Maybe out of the scope of this PR but why are all these tests async? I thought this was only useful if you were testing asynchronous code

Reply:
I'm not sure, what do you mean?

Reply:
That they accept the `done Done` parameter, it's an optional feature of Ginkgo `It`: https://onsi.github.io/ginkgo/#asynchronous-tests

Reply:
Not 100% sure, I guess they can all run in parallel